spring-context/src/test/java/org/springframework/context/annotation/ImportSelectorTests.java

```diff
@@ -203,6 +204,66 @@ void invokeAwareMethodsInImportGroup() {
 		assertThat(TestImportGroup.environment).isEqualTo(context.getEnvironment());
 	}
 
+	@Test
+	void importAnnotationOnImplementedInterfaceIsRespected() {
+		AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(
+				InterfaceBasedConfig.class);
+
+		assertThat(context.getBean(ImportedConfig.class)).isNotNull();
+		assertThat(context.getBean(ImportedBean.class)).isNotNull();
+		assertThat(context.getBean(ImportedBean.class).name()).isEqualTo("imported");
+	}
+
+	@Test
+	void localImportShouldOverrideInterfaceImport() {
+		AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(
+				OverridingConfig.class);
+
+		assertThat(context.getBean(ImportedConfig.class)).isNotNull();
+		assertThat(context.getBean(ImportedBean.class)).isNotNull();
+		assertThat(context.getBean(ImportedBean.class).name()).isEqualTo("from class");
+	}
+
+	@Import(ImportedConfig.class)
+	interface ConfigImportMarker {
+	}
+
+	@Configuration
+	static class InterfaceBasedConfig implements ConfigImportMarker {
+	}
+
+	@Configuration
+	static class OverridingConfig implements ConfigImportMarker {
```

Comment:
`OverridingConfig` should be annotated with `@Import` (and import something that overrides the `ImportedBean`) in order for `localImportShouldOverrideInterfaceImport()` to test this use case.